tests/Unit/Numerical/LinearAlgebra/Test_InvertMatrix.cpp

```diff
@@ -0,0 +1,172 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Numerical/LinearAlgebra/InvertMatrix.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+namespace {
+using generic = tmpl::integral_list<int32_t, 2, 1>;
+using symmetric = tmpl::integral_list<int32_t, 1, 1>;
+
+template <size_t Dim>
+using matrix_ii = Tensor<double, symmetric,
+                         index_list<SpatialIndex<Dim, UpLo::Lo, Frame::Grid>,
+                                    SpatialIndex<Dim, UpLo::Lo, Frame::Grid>>>;
+
+template <size_t Dim>
+using matrix_ij = Tensor<double, generic,
+                         index_list<SpatialIndex<Dim, UpLo::Lo, Frame::Grid>,
+                                    SpatialIndex<Dim, UpLo::Lo, Frame::Grid>>>;
+
+template <size_t Dim>
+using matrix_iA =
+    Tensor<double, generic,
+           index_list<SpatialIndex<Dim, UpLo::Lo, Frame::Grid>,
+                      SpatialIndex<Dim, UpLo::Up, Frame::Inertial>>>;
+
+// Checks that the inverse of a Tensor<double, Symm, index_list<I1, I2>>
+// is a Tensor<double, Symm, index_list<I3, I4>.
```

Comment:
Here's what I could replace it with:
```
auto inv = inverse(orig_t{}, det_t{});
static_assert(cpp17::is_same_v<inv_t, decltype(inv)>, "");
```

This reduces the complexity associated with getting the function's return type, but introduces the complexity of creating a temporary object. I have no strong preference.